roqoqo-iqm/src/devices/demo_device.rs

```diff
@@ -19,16 +19,27 @@ use std::cmp::{max, min};
 /// Provides endpoint that receives instructions and returns simulated results.
 /// Results are pseudo-random numbers, not actual quantum simulations.
 #[derive(Debug, PartialEq, Eq, Clone, serde::Serialize, serde::Deserialize)]
-pub struct DemoDevice;
+pub struct DemoDevice {
+    url: String,
+    number_qubits: usize,
+}
 
 impl DemoDevice {
     /// Create new DemoDevice
-    pub fn new() -> Self {
-        Self {}
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint_url` - The address of the REST API endpoint for the device
+    /// * `number_qubits` - The number of qubits of the device
+    pub fn new(endpoint_url: String, number_qubits: usize) -> Self {
+        Self {
+            url: endpoint_url,
+            number_qubits,
+        }
```

Comment:
Same as for the Python interface, we should add a separate `set_endpoint_url` funciton.